Hl7 2 ajp upload file to server

Perry5 opened the pull request:
Ticket: https://jira.amida.com/browse/HL7-2

## What does this ticket do?
- Adds endpoint to upload a file (using multer)
- Moves sample HL7 messages to `/data` folder
- Removes JWT requirement from config file
- Adds unit test for upload function
- Note that this doesn't do any validation of the file that is uploaded, that is a separate [ticket](https://jira.amida.com/browse/HL7-11). This only uploads the file and saves it. It was mainly to create a template so other people can work around it. 

## How to Test:
 -  Send a POST request from postman with a file in the request body
 - Make sure the file gets uploaded to `data/hl7-uploads`
 - Make sure your mongoDB has the name of the file and the date/time it was uploaded
- Make sure `yarn test` passes and there are no linting errors.

Perry5 committed:
rough draft of the upload function

Perry5 committed:
moved sample hl7 file to data folder

Perry5 committed:
added test for hl7 file upload

Perry5 committed:
modified test file

Perry5 committed:
fixed test

Perry5 committed:
added hl7-upload folder

orndorffgrant commented on package.json:
```diff
@@ -58,6 +58,7 @@
     "method-override": "^2.3.10",
     "mongoose": "4.7.4",
     "morgan": "1.9.1",
+    "multer": "false1.4.1",
```
Run `yarn config set save-prefix ""` so that yarn won't add `false` before version numbers

orndorffgrant commented on data/hl7-uploads/500HL7Messages.txt-1561400237743:
```diff
@@ -0,0 +1,5366 @@
+MSH|^~\&|App3|Facility2|App4|Facility6|201906191028||OMP^O09|3926758|D^I|2.2
```
I don't think the files in `hl7-uploads` should be stored in git

Perry5 replied:
Good point- let me fix that. Oversight on my part. Let me fix that

orndorffgrant commented on server/hl7/hl7.controller.js:
```diff
@@ -0,0 +1,36 @@
+
+const storage = multer.diskStorage({
+  destination: (req, file, cb) => { cb(null, 'data/hl7-uploads'); },
+  filename: (req, file, cb) => (cb(null, `${file.originalname}-${Date.now()}`))
```
this adds the date after the file extension, i.e. `originalfilename.txt-123456789`, not a big deal, but we probably want the file extension to stay at the end of the name

Perry5 replied:
Yes, this is intentional for now. Just wanted to create something for other to work from. Changes to this will be coming in another PR.

orndorffgrant commented on server/hl7/hl7.route.js:
```diff
@@ -0,0 +1,10 @@
+
+router.route('/upload')
+  /** POST /api/hl7/upload - Upload new HL7 document */
+  .post(hl7Ctrl.upload.single('hl7-message'), hl7Ctrl.uploadFile);
```
this is cool

orndorffgrant reviewed (changes requested):
Looks good! Couple small requests after code review. Functional review is next on my ToDo.

Perry5 committed:
ignoring everything in the data folder

Perry5 committed:
removed the sample hl7 messages

Perry5 committed:
removed false from multer

Perry5 committed:
removed uploaded files

Perry5 committed:
exclude everything in /data/

Perry5 commented:
@orndorffgrant updated

Commit:
cleaning up #2

- adding debug launch.json config for vscode
- adding back hl7 sample data
- keeping hl7-uploads directory but still not keeping any files placed there
- updating yarn.lock to align with package.json

orndorffgrant reviewed (approved):
Added back the hl7 sample data and kept the uploads directory without the uploads themselves.
Also added debug config for vscode.
Good to go!

Perry5 commented:
Rule of thumb is to have 2 approvers before we merge in a PR. However because this PR handles things like removing JWT and such, i'm going to merge it in so we are not dealing with those annoying issues.
